### ISSUE TITLE
lsc-core with spring boot version 2.2.2.RELEASE or above

### ISSUE THREAD

**Issue by Somyarajff** [lsc-project/lsc#173]
I am in the process of updating the spring version. It was 2.0.7.RELEASE and now I am at 2.2.1.RELEASE.
As soon as I update the version to 2.2.2.RELEASE or above I get the following error. 
java.lang.reflect.MalformedParameterizedTypeException: null
        at java.base/sun.reflect.generics.reflectiveObjects.ParameterizedTypeImpl.validateConstructorArguments(ParameterizedTypeImpl.java:59)
        at java.base/sun.reflect.generics.reflectiveObjects.ParameterizedTypeImpl.<init>(ParameterizedTypeImpl.java:52)
        at java.base/sun.reflect.generics.reflectiveObjects.ParameterizedTypeImpl.make(ParameterizedTypeImpl.java:93)
        at java.base/sun.reflect.generics.factory.CoreReflectionFactory.makeParameterizedType(CoreReflectionFactory.java:105)
        at java.base/sun.reflect.generics.visitor.Reifier.visitClassTypeSignature(Reifier.java:140)
        at java.base/sun.reflect.generics.tree.ClassTypeSignature.accept(ClassTypeSignature.java:49)
        at java.base/sun.reflect.generics.repository.ConstructorRepository.computeParameterTypes(ConstructorRepository.java:111)
        at java.base/sun.reflect.generics.repository.ConstructorRepository.getParameterTypes(ConstructorRepository.java:87)
        at java.base/java.lang.reflect.Executable.getGenericParameterTypes(Executable.java:279)
        at java.base/java.lang.reflect.Method.getGenericParameterTypes(Method.java:330)
        at java.desktop/java.beans.FeatureDescriptor.getParameterTypes(FeatureDescriptor.java:391)
        at java.desktop/java.beans.MethodDescriptor.setMethod(MethodDescriptor.java:118)
        at java.desktop/java.beans.MethodDescriptor.<init>(MethodDescriptor.java:74)
        at java.desktop/java.beans.MethodDescriptor.<init>(MethodDescriptor.java:58)
        at java.desktop/java.beans.Introspector.getTargetMethodInfo(Introspector.java:1046)
        at java.desktop/java.beans.Introspector.getBeanInfo(Introspector.java:461)
        at java.desktop/java.beans.Introspector.getBeanInfo(Introspector.java:204)
        at groovy.lang.MetaClassImpl$15.run(MetaClassImpl.java:2924)
        at java.base/java.security.AccessController.doPrivileged(Native Method)
        at groovy.lang.MetaClassImpl.addProperties(MetaClassImpl.java:2922)
        at groovy.lang.MetaClassImpl.initialize(MetaClassImpl.java:2905)
        at org.codehaus.groovy.reflection.ClassInfo.getMetaClassUnderLock(ClassInfo.java:164)
        at org.codehaus.groovy.reflection.ClassInfo.getMetaClass(ClassInfo.java:180)
        at org.codehaus.groovy.runtime.metaclass.MetaClassRegistryImpl.getMetaClass(MetaClassRegistryImpl.java:195)
        at org.springframework.beans.factory.groovy.GroovyBeanDefinitionReader.<init>(GroovyBeanDefinitionReader.java:150)
        at org.springframework.boot.BeanDefinitionLoader.<init>(BeanDefinitionLoader.java:85)
        at org.springframework.boot.SpringApplication.createBeanDefinitionLoader(SpringApplication.java:738)
        at org.springframework.boot.SpringApplication.load(SpringApplication.java:681)
        at org.springframework.boot.SpringApplication.prepareContext(SpringApplication.java:392)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:314)
        at org.springframework.boot.builder.SpringApplicationBuilder.run(SpringApplicationBuilder.java:140)
        at org.springframework.cloud.bootstrap.BootstrapApplicationListener.bootstrapServiceContext(BootstrapApplicationListener.java:212)
        at org.springframework.cloud.bootstrap.BootstrapApplicationListener.onApplicationEvent(BootstrapApplicationListener.java:117)
        at org.springframework.cloud.bootstrap.BootstrapApplicationListener.onApplicationEvent(BootstrapApplicationListener.java:74)
        at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:172)
        at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:165)
        at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:139)
        at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:127)
        at org.springframework.boot.context.event.EventPublishingRunListener.environmentPrepared(EventPublishingRunListener.java:76)
        at org.springframework.boot.SpringApplicationRunListeners.environmentPrepared(SpringApplicationRunListeners.java:53)
        at org.springframework.boot.SpringApplication.prepareEnvironment(SpringApplication.java:345)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:308)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1226)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1215)
        at com.microservices.SystemApiApplication.main(SystemApiApplication.java:65)

The dependency in pom.xml file is - 
```
<dependency>
	<groupId>org.lsc</groupId>
	<artifactId>lsc-core</artifactId>
	<version>2.1.6</version>
	<exclusions>
		<exclusion>
			<groupId>org.forgerock.opendj</groupId>
			<artifactId>opendj-server</artifactId>
		</exclusion>
	</exclusions>
</dependency>

<repositories>
    <repository>
	    <id>lsc-site</id>
	    <url>http://lsc-project.org/maven</url>
    </repository>
</repositories>
```
If I comment the dependency I am able to successfully build the application but in that case my functionality which uses lsc library is stopped. Are you guys aware of any issue like this?

**Comment by soisik** [comment]
Seems to be related to groovy. lsc  2.1.6 is using quite an old version of groovy, we have recently updated it to 2.4.21, but it will not be available until 2.1.7. 

Maybe try to exclude this dependency from LSC

```<exclusion>
       <groupId>org.codehaus.groovy</groupId>
       <artifactId>groovy-all</artifactId>                  
</exclusion>```

It could be necessary to also declare a dependency to groovy if not already present (version will be handled by spring boot according to https://docs.spring.io/spring-boot/docs/2.2.1.RELEASE/reference/html/appendix-dependency-versions.html#appendix-dependency-versions)
```
<dependency>
        <groupId>org.codehaus.groovy</groupId>
        <artifactId>groovy</artifactId>
</dependency>
```

If it doesn't work, could you provide your pom.xml in order to reproduce the issue ?

**Comment by Somyarajff** [comment]
Exclusion worked for me. Thanks for your help.
